lib/URL-impl.js

```diff
@@ -19,9 +20,12 @@ exports.implementation = class URLImpl {
       throw new TypeError("Invalid URL");
     }
 
+    const query = parsedURL.query || "";
+
     this._url = parsedURL;
 
-    // TODO: query stuff
+    this._query = URLSearchParams.createImpl([query], { doNotStripQMark: true });
```

Comment:
"doNotStripQMark" does not appear to be necessary in the spec. Why do we have it here?

In general I would think we could just parallel the spec more closely for construction, but I might be wrong...